frontend/src/libs/components/loader/loader.tsx

```diff
@@ -3,21 +3,38 @@ import { getValidClassNames } from '~/libs/helpers/helpers.js';
 import styles from './styles.module.scss';
 
 type Properties = {
-  children: React.ReactNode;
+  children?: React.ReactNode;
   className?: string;
   isLoading: boolean;
+  hasOverlay?: boolean;
+  loaderType: 'dots' | 'circular';
 };
 
-const Loader: React.FC<Properties> = ({ children, className, isLoading }) => {
-  if (!isLoading) {
-    return children;
+const Loader: React.FC<Properties> = ({
+  children,
+  className,
+  isLoading,
+  hasOverlay = false,
+  loaderType,
+}) => {
+  if (isLoading) {
+    const containerClassName = hasOverlay
+      ? styles.overlayContainer
+      : styles.notOverlayContainer;
+
+    const loaderComponentClassName =
+      loaderType === 'circular' ? styles.circularLoader : styles.dotFlashing;
+
+    return (
+      <div className={containerClassName}>
+        <div
+          className={getValidClassNames(loaderComponentClassName, className)}
+        ></div>
```

Comment:
As we've restricted `loaderType` value with predefined values - `circular` | `dots`
Let's just bring styling classes with same names and make it easier:
```suggestion

    return (
      <div className={containerClassName}>
        <div className={getValidClassNames(styles[loaderType], className)} />
```